eureka-client-archaius2/src/main/java/com/netflix/appinfo/Ec2EurekaArchaius2InstanceConfig.java

```diff
@@ -75,7 +80,10 @@ else if ((amazonInfo.get(MetaDataKey.publicHostname) == null)
             // public hostname might be null
             amazonInfo.getMetadata().put(MetaDataKey.publicHostname.getName(),
                     (amazonInfo.get(MetaDataKey.localIpv4)));
-        }        
+        }
+
+        // TODO: Remove this when DiscoveryManager is finally no longer used
+        DiscoveryManager.getInstance().setEurekaInstanceConfig(this);
```

Comment:
Replacement is to not use the Manager and just create things either via DI or directly.